spec/models/instance_spec.rb

```diff
@@ -1,6 +1,29 @@
 require 'rails_helper'
 
 RSpec.describe Instance, type: :model do
+  let(:instance) { build(:instance) }
+
+  context 'when host format is invalid' do
+    it 'is expected to be invalid' do
+      hosts = ['example,com', 'www_.example.org', 'example_.org',
+               'example.org_', 'example..org', 'example.org.']
+      hosts.each do |invalid_host|
+        instance.host = invalid_host
+        expect(instance).not_to be_valid
+      end
+    end
+  end
+
+  context 'when host format is valid' do
+    it 'is expected to be valid' do
+      hosts = ['example.ORG', 'www.example.org', 'ex-ample.org', 'example.org.sg']
```

Comment:
I think the each should be outside the test. then for each you actually have one separate test case...